File: src/backend/core/shared/models/agent.py
"""Agent 领域模型。"""

from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from typing import Sequence


@dataclass
class ToolDefinition:
    """工具定义领域对象。"""

    id: str
    name: str
    description: str
    schema: dict


@dataclass
class Agent:
    """Agent 领域对象。"""

    id: str
    owner_id: str
    name: str
    description: str
    system_prompt: str
    model: str
    tool_ids: Sequence[str] = field(default_factory=list)
    status: str = "idle"
    created_at: datetime | None = None
    updated_at: datetime | None = None

    def is_owned_by(self, user_id: str) -> bool:
        """检查 Agent 是否属于指定用户。"""
        return self.owner_id == user_id

File: frontend-public/lib/types/agent.ts
export type AgentStatus = "idle" | "running" | "error" | "offline"

export interface Agent {
  id: string
  owner_id: string
  name: string
  description: string
  system_prompt: string
  model: string
  tool_ids: string[]
  status: AgentStatus
  created_at?: string
  updated_at?: string
}

export interface AgentCreatePayload {
  name: string
  description?: string
  system_prompt: string
  model: string
  tool_ids?: string[]
}

export interface AgentUpdatePayload extends AgentCreatePayload {
  status?: AgentStatus
}
